internal/templates/commands/render_test.go

```diff
@@ -0,0 +1,217 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/abcxyz/pkg/testutil"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestParseFlags(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name    string
+		args    []string
+		want    flagValues
+		wantErr string
+	}{
+		{
+			name: "all flags present",
+			args: []string{
+				"--template", "helloworld@v1",
+				"--spec", "my_spec.yaml",
+				"--dest", "my_dir",
+				"--git-protocol", "https",
+				"--input", "x=y",
+				"--log-level", "info",
+				"--force-overwrite",
+				"--keep-temp-dirs",
+			},
+			want: flagValues{
+				template:       "helloworld@v1",
+				spec:           "my_spec.yaml",
+				dest:           "my_dir",
+				gitProtocol:    "https",
+				inputs:         map[string]string{"x": "y"},
+				logLevel:       "info",
+				forceOverwrite: true,
+				keepTempDirs:   true,
+			},
+		},
+		{
+			name: "minimal flags present",
+			args: []string{
+				"-t", "helloworld@v1",
+			},
+			want: flagValues{
+				template:       "helloworld@v1",
+				spec:           "./spec.yaml",
+				dest:           ".",
+				gitProtocol:    "https",
+				inputs:         nil,
+				logLevel:       "warning",
+				forceOverwrite: false,
+				keepTempDirs:   false,
+			},
+		},
+		{
+			name:    "a required flag is missing",
+			args:    []string{},
+			wantErr: "-template is required",
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			c := &Render{}
+			err := c.parseFlags(tc.args)
+			if tc.wantErr != "" {
+				if diff := testutil.DiffErrString(err, tc.wantErr); diff != "" {
+					t.Fatal(diff)
+				}
+				return
+			}
+
+			if diff := cmp.Diff(c.fv, tc.want, cmp.AllowUnexported(flagValues{})); diff != "" {
+				t.Errorf("got %#v, want %#v, diff (-got, +want): %v", c.fv, tc.want, diff)
+			}
+		})
+	}
+}
+
+func TestDestOK(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name    string
+		flags   *flagValues
+		in      fakeFS
```

Comment:
Have you looked at https://pkg.go.dev/testing/fstest?

Reply:
Done, thanks.